algorithms/src/main/kotlin/com/kotlinground/algorithms/stack/minstrlen/README.md

```
@@ -0,0 +1,219 @@
+position indicated by writePtr. Next, we check if the last two characters (positions writePtr-1 and writePtr) form a
+removable pattern, such as "AB" or "CD". If they do, we decrease writePtr, which means we will overwrite this part in
+the next steps.
+
+If the last two characters do not form a removable pattern, we increment writePtr to point to the next position where
+we can write a character. This way, we can effectively "remove" unwanted characters by overwriting them.
+
+After processing all characters, the position of writePtr tells us the length of the final string. All the characters
+we want to keep are now at the beginning of the array, up to the position of writePtr.
+
+![Solution 1](./images/solutions/minimum_string_length_after_removing_substrings_in-place_solution_1.png)
+![Solution 2](images/solutions/minimum_string_length_after_removing_substrings_in-place_solution_1.png)
```

markdownlint-cli2 (0.21.0)

[warning] Spaces inside emphasis markers

(MD037, no-space-in-emphasis)